reactioncommerce/reaction
# chore|style: ESLint errors and warnings === 0

Resolves #5284, #5276, #5267, #5269   
Impact: **major**  
Type: **chore|style**

This PR:
1. Fixes the remaining `eslint` warnings in our codebase, where applicable
1. Disables warnings and creates tickets for `eslint` issues that would require a bit more refactoring / thought, such as `react/no-did-update-set-state`, which will be fixed when components are updated to be functional components.
1. Removes all `warning` overrides in our `package.json` file, which now means **_every eslint rule we follow is now an error_**.
1. Removes the `eslint-changed-files` CI step, which is no longer needed since every rule is an error, and our classic `eslint` CI check takes care of all errors.

## Breaking changes
None for our codebase. Any custom code might need to be updated to obey the new lint rules.

## Testing
1. See that the app starts up
1. See that there are 0 errors and 0 warnings when running `eslint`
1. See that there are no longer any `warn` overrides for `eslint` rules in our package.json
1. See that there is no longer the `eslint-changed-files` CI check run on this PR

![1__reaction__bash_](https://user-images.githubusercontent.com/4482263/62981044-15a3f700-bddd-11e9-91c1-8171e879393a.gif)
